fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SplitHelper.java

```diff
@@ -665,6 +671,12 @@ public KeyValueUnsplitter(@Nonnull FDBRecordContext context, @Nonnull final Subs
 
                     if (next == null) { // no next result
                         nextResult = RecordCursorResult.withoutNextValue(continuation, mergeNoNextReason());
+                        if (LOGGER.isTraceEnabled()) {
```

Comment:
Just curious as to when we get into this code. Is the inner cursor fully exhausted at this point?

Reply:
It usually should be stopped, yes. (Which is different from exhausted, at least insofar as it is used as term of art within the record layer.) This log message is included so that we can verify that (it's designed to, for example, test that there isn't something wonky going on with one of the while loops).